### ISSUE TITLE
Failed to dectect apple as vendor for x86_64-apple-darwin

### ISSUE THREAD

**Issue by weiznich** [nagisa/target_build_utils.rs#1]
I've trying to port the [build system of ring](https://github.com/briansmith/ring/pull/321) from makefiles to a build.rs based system.
To better handle the the detection for which target ring is build, I use this crate. It's basically working, but there are strange ci-failures on macos. It seems like target_build_utils is unable to detect [apple as vendor](https://travis-ci.org/briansmith/ring/jobs/192684642#L1307) even if [target](https://travis-ci.org/briansmith/ring/jobs/192684642#L367) is set to  x86_64-apple-darwin. I'm not sure why this happen, there seems to be an [test](https://github.com/nagisa/target_build_utils.rs/blob/master/src/lib.rs#L328) in this crate that also passes on my local machine.

**Comment by nagisa** [comment]
This crate essentially parses the output of `rustc --print=cfg --target=x86_64-apple-darwin`. Vendor missing either means I messed up badly writing the [trivial parser](https://github.com/nagisa/target_build_utils.rs/blob/master/build.rs#L69), or the rustc simply prints incomplete cfg, like this:

```
$ rustup run stable rustc --print=cfg --target=x86_64-apple-darwin
target_os="macos"
target_family="unix"
target_arch="x86_64"
target_endian="little"
target_pointer_width="64"
target_env=""
unix
debug_assertions
▪rustup run stable rustc --version
rustc 1.14.0 (e8a012324 2016-12-16)
```

I’ll think what can be done about this.

**Comment by nagisa** [comment]
As a workaround, its probably best to check `target_os == "macos" || target_os == "ios"` instead.

**Comment by nagisa** [comment]
see https://github.com/rust-lang/rust/issues/29718.

**Comment by nagisa** [comment]
Thanks for the report. I’ve published version 0.2.0 which contains https://github.com/nagisa/target_build_utils.rs/commit/d337924b67018398a8df0e5c0258d0adb96ae927 and makes `target_vendor` to return `Option`.
